src/camera_positioner.cpp

```diff
@@ -65,11 +66,14 @@ class CameraPositioner {
         if(msg.detections[i].id[0] == 0){
            tf::Transform tag_transform;
            tf::poseMsgToTF(msg.detections[i].pose.pose.pose, tag_transform);
-           world_camera_transform= world_tag_transform * tag_transform.inverse() * optical_transform;
            if(!initialized){
               ROS_INFO("camera positioner is running");
               initialized = true;
+           } else {
+              interpolateTransforms(last_tag_transform, tag_transform, 0.05, tag_transform);
```

Comment:
Please make the 0.05 a variable above, maybe even add a rosparameter for it.
There's nothing worse than changing magic parameters in function calls in the middle of some code to adjust something...